### DISCUSSION TITLE
Phase 3 - Alpha

### DISCUSSION THREAD

**Discussion by alvarix** [alvarix/FLogger#90] [Ideas]
[is:issue is:open -label:legacy assignee:alvarix ](https://github.com/alvarix/FLogger/issues?q=is%3Aissue+is%3Aopen+-label%3Alegacy+assignee%3Aalvarix)

### Task:
- Finish **_editEntry()_**
  - Discussed different strategies for edit entry, mimicking deleteEntryFromFlog in useFlogs composable.
    - Done:
      - save on blur; revert to div on blur (no textarea)
      - unearthed dbx save bug (in PR) 
      - auto grow textarea to size of entry (may not be possible)
  - To do:
    - also change Edit button to Save and trigger save function?
    
### Next tasks:
- On _closeflog_ list of flogs should still be visible
- Display Dropbox Account
- Dark/Light mode styles
- deleteFlog

### Task:
- **Vercel URLs**

### Next tasks:
- **dbx save bug?**
- feedback to **AddFlog** 
  - add label 'add title' to input (file ext instructions)
    - on Submit add flog button, open flog and focus _AddEntry_ 
  - openFlogs immediately updated with new flog
- **Starter files + Prefs**
  - Discussed Sources and keeping sourceType as an extensor to add other platforms in the future (eg Google Files).
That in turn brought up the idea of a starter flogs mechanism (eg. README.flogger.txt, example-flog.flogger.txt) and a prefs files that get copied into the users' dropbox folder on init.

**Comment by alvarix** [comment]
## Future Ideas
- Right panel 
  - **Prefs** 
    - eg: Dont show README and example file
  - **available Flogs** list
    - options:
      - replace open Flog
      - blend with open Flog
      - open next to open Flog with dates aligned 
- UX
  - when text in editor and do a function - **confirm('Do you want to save work?')**
- **rename file**

**Comment by alvarix** [comment]
@chadcrume i thought of another pattern for add file. 
It goes back to mimicking nvalt. 

There’s a search bar at top that you type into - if it matches an existing file (potentially with auto complete) then it opens that file for editing. Else it creates a new file. 
wdyt?

**Comment by alvarix** [comment]
@chadcrume 
started new issues. 
Filter out the old stuff with:
https://github.com/alvarix/FLogger/issues?q=is%3Aissue+is%3Aopen+-label%3Alegacy+

**Comment by alvarix** [comment]
@chadcrume Im finding that prop drilling is confusing me. 
Curious if you think Pinia would make things easier?
Or does that obviate some of your work creating composables.

[9 Ways of Component Communication in Vue3](https://www.dropbox.com/scl/fi/v9qturhrv36dv1d862rvt/9-Ways-of-Component-Communication-in-Vue3-_-by-Shawn-King-Freedium.pdf?rlkey=6jsjwagjysiuzomf4dkywwdap&dl=0)

**Reply by alvarix** [reply]
@chadcrume

**Comment by alvarix** [comment]
@chadcrume tried out the vercel page. 
Couple of issues:

1. chokes on certain chars? 
[prx1.3.flogger.txt](https://github.com/user-attachments/files/17667153/prx1.3.flogger.txt)
After opening the above file it just shows one entry with NaN/NaN/NaN.
How practical is it to address this?
2. DBX cache? 
I tried to address the above by deleting the non-dat chars, but I cant get flogger to see the file changes. 
I duped the file and removed the chars about 20m ago - neither the changes or new file are appearing in file list.

**Reply by alvarix** [reply]
@chadcrume 
damn still getting NaN/NaN/NaN on my file in the vercel testing URL.
I assumed it was the frontmatter causing it but Ive removed it [in this file](https://www.dropbox.com/scl/fi/7e5ubc0t1yj4dzo04g54a/prx1.3.flogger.txt?rlkey=vehknkdhjndkgb4gldsr1b4u5&dl=0)
Do we need a new testing URL?

**Reply by alvarix** [reply]
@chadcrume disregard. Bad date format!

**Comment by chadcrume** [comment]
What's your testing URL?

**Reply by alvarix** [reply]
@chadcrume fyi i added 

- [ ]  formatting error handlers? (bad dates)

to both our lists, whoever gets there first and has the will
